components/work_templates/index.tsx

```diff
@@ -170,6 +191,85 @@ const WorkTemplateModal = () => {
+        const res = {
+            playbooks: 0,
+            boards: 0,
+            jira: 0,
+            github: 0,
+            gitlab: 0,
+        };
+        for (const item of template.content) {
+            if (item.playbook) {
+                const pbTemplate = playbookTemplates.find((pb) => pb.title === item.playbook.template);
+                if (pbTemplate) {
+                    res.playbooks++;
+                }
+            } else if (item.board) {
+                res.boards++;
+            } else if (item.integration) {
+                if (item.integration.id === PLUGIN_NAME_TO_ID_MAP.jira) {
+                    res.jira = 1;
+                } else if (item.integration.id === PLUGIN_NAME_TO_ID_MAP.github) {
+                    res.github = 1;
+                } else if (item.integration.id === PLUGIN_NAME_TO_ID_MAP.gitlab) {
+                    res.gitlab = 1;
+                }
+            }
+        }
+
+        return res;
+    };
+
+    /**
+     * Creates the necessary data in the global store as long storing in DB preferences the tourtip information
+     * @param template current used worktempplate
+     */
+    const tourTipActions = async (template: WorkTemplate) => {
+        const linkedProductsCount = getContentCount(template);
+
+        // stepValue and pluginId are used for showing the tourtip for the used template
+        let stepValue = 0;
+        let pluginId;
+        if (linkedProductsCount.playbooks) {
+            pluginId = rhsPluggableIds.get(suitePluginIds.playbooks);
+            stepValue = WorkTemplateTourSteps.PLAYBOOKS_TOUR_TIP;
+        } else {
+            pluginId = rhsPluggableIds.get(suitePluginIds.boards);
+            stepValue = WorkTemplateTourSteps.BOARDS_TOUR_TIP;
+        }
+
+        if (pluginId) {
+            dispatch(showRHSPlugin(pluginId));
+        }
+
+        // store the required preferences for the tourtip
+        const tourCategory = TutorialTourName.WORK_TEMPLATE_TUTORIAL;
+        const preferences = [
+
+            // here reset the step value to be able to show the tour again (if we dedide to show the tour only once, this must be removed)
+            {
+                user_id: currentUserId,
+                category: tourCategory,
+                name: currentUserId,
+                value: stepValue.toString(),
+            },
+
+            // this one is for defining the auto tour start for the tour tip
+            {
+                user_id: currentUserId,
+                category: tourCategory,
+                name: TTNameMapToATStatusKey[tourCategory],
+                value: String(AutoTourStatus.ENABLED),
+            },
+        ];
+        await dispatch(savePreferences(currentUserId, preferences));
+
+        // store in the global state the plugins/integrations information related to the used template
+        // so we can display that data in the tourtip
+        await dispatch(onExecuteSuccess(linkedProductsCount));
```

Comment:
Is it important that `onExecuteSuccess` waits for `savePreferences` to complete? If not, I think we can get rid of the `async` on this function because nothing else within it waits, and it looks like where it is called, it is not awaited either.

Reply:
right, no need to await for it